pytket/extensions/quantinuum/backends/quantinuum.py

```diff
@@ -411,6 +423,7 @@ def default_compilation_pass(self, optimisation_level: int = 2) -> BasePass:
                     SimplifyInitial(
                         allow_classical=False, create_all_qubits=True, xcirc=_xcirc
                     ),
+                    CustomPass(flatten_registers),
```

Comment:
Since this is done at all optimization levels, can we just add it unconditionally that the end instead of inside each branch?

Reply:
Looks like it is still being added in the individual branches?

Reply:
Oops, done.